session.go

```diff
@@ -713,14 +729,18 @@ func (q Query) String() string {
 }
 
 //Attempts returns the number of times the query was executed.
-func (q *Query) Attempts() int {
-	return q.attempts
+func (q *Query) Attempts(host *HostInfo) uint64 {
```

Comment:
The main gripe I have with this approach is that this a) break backwards compatibility in a major way and b) now produces a different behavior even with the default `SimpleRetryPolicy`. The metrics are tracked by host and default retry type of the `SimpleRetryPolicy` is to `RetryNextHost` which means the attempts are now also _per host_. The max retries are ~therefor also `number of hosts * max retries - (number of hosts - 1)` (don't ask, the gist: it fills~ (that's not correct) not honored because on each successive retry (on the next host) the attempts for that will be 0 and it will exhaust the host pool and fail with a different error than before. Mind you, I'm not harking on the general direction or the feature (it _is_ very useful) but maybe leave the current signatures alone and keep track of a summed version of attempts and latencies (as it is right now) and make the per host latencies and attempts available to the retry policy through some other means so going forward one can create an advanced retry policy that takes those into account? 🤔

Reply:
Hmmm...now I'm kinda confused.

So, you're right, the current `queryExecution` only checks that the total accumulated attempts across all hosts have not surpassed the number set in the retry policy, and this approach changes that. I was wrong in my understanding of what it does.

So...I think I understand it now. What I will probably do is introduce a way to find the total count by summing the map values; also will see how to keep signatures the same.

Thanks for the review!